runtime/src/dpe_crypto.rs

```diff
@@ -78,21 +96,55 @@ impl<'a> Crypto for DpeCrypto<'a> {
 
     fn derive_cdi(
         &mut self,
-        _algs: AlgLen,
-        _measurement: &Digest,
-        _info: &[u8],
+        algs: AlgLen,
+        measurement: &Digest,
+        info: &[u8],
     ) -> Result<Self::Cdi, CryptoError> {
-        Err(CryptoError::NotImplemented)
+        match algs {
+            AlgLen::Bit256 => Err(CryptoError::Size),
+            AlgLen::Bit384 => {
+                let mut hmac_key = [0u8; AlgLen::Bit384.size()];
+                hmac_key.copy_from_slice(measurement.bytes());
+                hmac384_kdf(
+                    self.hmac384,
+                    Hmac384Key::Array4x12(&Array4x12::from(hmac_key)),
+                    b"derive_cdi",
+                    Some(info),
+                    &mut self.trng,
+                    KeyWriteArgs::new(KEY_ID_RT_CDI, KeyUsage::default().set_hmac_data_en()).into(),
```

Comment:
A few things here

* I think the input key should be `KEY_ID_RT_CDI`. We want to derive a CDI based on that
* The context argument can contain both the `info` and `measurement` parameters. The most convenient thing to do may be to hash them together and pass the output digest as the context.
* For the output CDI, we'll want to claim a keyid that isn't already being used by ROM/FMC.